// angular-auth-oidc-client/esm2020/lib/extractors/jwk.extractor.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class JwkExtractor {
    static buildErrorName(name) {
        return JwkExtractor.name + ': ' + name;
    }
    extractJwk(keys, spec, throwOnEmpty = true) {
        if (0 === keys.length) {
            throw JwkExtractor.InvalidArgumentError;
        }
        let foundKeys = keys
            .filter((k) => spec?.kid ? k['kid'] === spec.kid : true)
            .filter((k) => spec?.use ? k['use'] === spec.use : true)
            .filter((k) => spec?.kty ? k['kty'] === spec.kty : true);
        if (foundKeys.length === 0 && throwOnEmpty) {
            throw JwkExtractor.NoMatchingKeysError;
        }
        if (foundKeys.length > 1 && (null === spec || undefined === spec)) {
            throw JwkExtractor.SeveralMatchingKeysError;
        }
        return foundKeys;
    }
}
JwkExtractor.InvalidArgumentError = {
    name: JwkExtractor.buildErrorName('InvalidArgumentError'),
    message: 'Array of keys was empty. Unable to extract'
};
JwkExtractor.NoMatchingKeysError = {
    name: JwkExtractor.buildErrorName('NoMatchingKeysError'),
    message: 'No key found matching the spec'
};
JwkExtractor.SeveralMatchingKeysError = {
    name: JwkExtractor.buildErrorName('SeveralMatchingKeysError'),
    message: 'More than one key found. Please use spec to filter'
};
JwkExtractor.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.0", ngImport: i0, type: JwkExtractor, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
JwkExtractor.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.1.0", ngImport: i0, type: JwkExtractor });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.0", ngImport: i0, type: JwkExtractor, decorators: [{
            type: Injectable
        }] });
